docs/sources/tempo/traceql/trace-structure.md

```diff
@@ -0,0 +1,83 @@
+
+## Trace structure
+
+Traces are telemetry data structured as trees.
+Traces are made of spans (for example, a span tree); there is a root span that can have zero to multiple branches that are called child spans.
+Each child span can itself be a parent span of one or multiple child spans, and so on so forth.
+
+![Trace_and_spans_in_tree_structure](/media/docs/tempo/traceql/trace-tree-structures-and-spans.png)
+
+In the specific context of TraceQL, a span has the following associated fields:
+
+- **name**: the span name
+- **duration**: difference between the end time and start time of the span
+- **status**: enum: `{ok, error, unset}`. For details, refer to [OTel span status](https://opentelemetry.io/docs/concepts/signals/traces/#span-status) documentation.
+- **kind**: enum: `{server, client, producer, consumer, internal, unspecified}`. For more details, refer to [OTel span kind ](https://opentelemetry.io/docs/concepts/signals/traces/#span-kind) documentation.
+- Attributes
+
+The first four properties are *intrinsics*.
+They are fundamental to the span.
+
+*Attributes* are custom span metadata in the form of key-value pairs.
+There are two types of attributes: **span attributes** and **resource attributes**.
+
+*Span attributes* are key-value pairs that contain metadata that you can use to annotate a span to carry information about the operation it's tracking.
+For example, in an eCommerce application, if a span tracks an operation that adds an item to a user’s shopping cart, the user ID, added item ID and cart ID can be captured and attached to the span as span attributes.
+
+A *Resource attribute* represents information about an entity producing the span.
+For example, a span created by a process running in a container deployed by Kubernetes could link a resource that specifies the cluster name, namespace, pod, and container name.
+Resource attributes are resource-related metadata (key-value pairs) that are describing the Resource.
+
+The OpenTelemetry specification defines Semantic Attributes for Spans and for Resources.
+Semantic Span Attributes are a set of naming schemes for attributes shared across languages, frameworks, and runtimes.
+For more details, refer to [Trace Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/general/trace/) and [Resource Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/resource/).
+
+## TraceQL queries
+
+The purpose of TraceQL is to search or query for spans.
+The query returns a set of spans, also called a spanset.
+
+A TraceQL query can select traces based on:
+
+- span attributes, timing and duration
+- structural relationships between spans
+- aggregated data from the spans in a trace
+
+As with PromQL and LogQL, the query is structured as a pipeline of operations (filters and aggregators).
+The query expression is evaluated on one trace at a time, selecting or discarding spans from the trace.
```

Comment:
"from the trace", might be "from the result".

Reply:
Done.